cli/standalone.rs

```diff
@@ -0,0 +1,114 @@
+use crate::colors;
+use crate::flags::Flags;
+use crate::permissions::Permissions;
+use crate::program_state::ProgramState;
+use crate::tokio_util;
+use crate::worker::MainWorker;
+use deno_core::error::AnyError;
+use deno_core::futures::FutureExt;
+use deno_core::ModuleLoader;
+use deno_core::ModuleSpecifier;
+use deno_core::OpState;
+use std::cell::RefCell;
+use std::convert::TryInto;
+use std::env::current_exe;
+use std::fs::File;
+use std::io::Read;
+use std::io::Seek;
+use std::io::SeekFrom;
+use std::pin::Pin;
+use std::rc::Rc;
+
+pub fn standalone() {
+  let current_exe_path =
+    current_exe().expect("expect current exe path to be known");
+
+  let mut current_exe = File::open(current_exe_path)
+    .expect("expected to be able to open current exe");
+  let magic_trailer_pos = current_exe
+    .seek(SeekFrom::End(-12))
+    .expect("expected to be able to seek to magic trailer in current exe");
+  let mut magic_trailer = [0; 12];
+  current_exe
+    .read_exact(&mut magic_trailer)
+    .expect("expected to be able to read magic trailer from current exe");
+  let (magic_trailer, bundle_pos) = magic_trailer.split_at(4);
+  if magic_trailer == b"DENO" {
+    let bundle_pos_arr: &[u8; 8] =
+      bundle_pos.try_into().expect("slice with incorrect length");
+    let bundle_pos = u64::from_be_bytes(*bundle_pos_arr);
+    current_exe
+      .seek(SeekFrom::Start(bundle_pos))
+      .expect("expected to be able to seek to bundle pos in current exe");
+
+    let bundle_len = magic_trailer_pos - bundle_pos;
+    let mut bundle = String::new();
+    current_exe
+      .take(bundle_len)
+      .read_to_string(&mut bundle)
+      .expect("expected to be able to read bundle from current exe");
+    // TODO: check amount of bytes read
```

Comment:
I also think we should hash embedded bundle and add that hash to the trailer to verify it when booting standalone binary

Reply:
I agree, but can we postpone that to when we add metadata like flags and import specifier to the self contained binary?

Reply:
Sure

Reply:
> I also think we should hash embedded bundle and add that hash to the trailer to verify it when booting standalone binary

Hashing is probably not going to help: one having permission to modify the file can tamper with the hash anyways (hashing or any other kind of auth is only useful if you could obtain the hash through a separate channel or encrypted), thus this layer of protection is unnecessary.